rskj-core/src/main/java/co/rsk/db/importer/provider/BootstrapDataProvider.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2019 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.db.importer.provider;
+
+import co.rsk.db.importer.BootstrapImportException;
+import co.rsk.db.importer.provider.index.BootstrapIndexMCH;
+import co.rsk.db.importer.provider.index.BootstrapIndexRetriever;
+import co.rsk.db.importer.provider.index.data.BootstrapDataEntry;
+import co.rsk.db.importer.provider.index.data.BootstrapDataIndex;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+
+public class BootstrapDataProvider {
+
+    private final BootstrapIndexMCH bootstrapIndexMCH;
+    private final BootstrapDataVerifier bootstrapDataVerifier;
+    private final BootstrapFileHandler bootstrapFileHandler;
+    private BootstrapIndexRetriever bootstrapIndexRetriever;
+    private int minimumAcceptableSources;
+    private long height;
+
+    public BootstrapDataProvider(
+            BootstrapDataVerifier bootstrapDataVerifier,
+            BootstrapFileHandler bootstrapFileHandler,
+            BootstrapIndexMCH bootstrapIndexMCH,
+            BootstrapIndexRetriever bootstrapIndexRetriever,
+            int minimumAcceptableSources) {
+
+        this.bootstrapIndexMCH = bootstrapIndexMCH;
+        this.bootstrapDataVerifier = bootstrapDataVerifier;
+        this.bootstrapFileHandler = bootstrapFileHandler;
+        this.bootstrapIndexRetriever = bootstrapIndexRetriever;
+        this.minimumAcceptableSources = minimumAcceptableSources;
```

Comment:
I think it would be interesting for `minimumAccetpableSources >= 3` to disallow misconfigurations which could lead to a corrupted database. 

Example scenario, not even an attack: a user messes with settings (not an impossible scenario for a settings file) and for some reason leaves only one public key on it. The BoostrapdataProvider would continue to function, but it would stop being a multisig scheme quite silently.

Reply:
I'm not a fan of stepping over a user configuration value, the amount of public keys configured will determine this value. Nevertheless I can add a validation for this number with a warning in log messages whenever the value is lower than 2 (the needed amount of sources when there are 2 or more public keys used).